### DISCUSSION TITLE
Register vs Controller refs

### DISCUSSION THREAD

**Discussion by fpaczkov** [react-hook-form/react-hook-form#7766] [Q&A]
Hi! 
It's my first post here. I encountered a problem while creating custom scrolling to errors:
![image](https://user-images.githubusercontent.com/15363014/152771223-07403866-ed84-4b70-a37d-98aa17d7a74a.png)
Refs of the first (field created using `register`) and second (field created using `controller`) are different and I am not able to get `scrollIntoView` method from the second one.
Can someone explain why they are different? How can I get seconds element regular ref while using `formState.errors`?

react-hook-form version: 7.21.2

**Comment by Moshyfawn** [comment]
What's the behavior you're trying to achieve?
RHF takes care of focus and scroll of the input with an error.

Would you mind looking into [this CSB](https://codesandbox.io/s/eager-grass-3kqb6?file=/src/App.tsx) and tweaking it to closer match your use-case?

**Reply by Moshyfawn** [reply]
Also, happy first post lol 🎂

**Reply by fpaczkov** [reply]
I am using standard inputs and custom fields (these I had problems with). I'll to adjust the CSB later.

**Reply by fpaczkov** [reply]
Well, I have found the cause of my problem why some fields are not focused by the default lib mechanism. I had to add `tabIndex` to root divs of my components. I think that it would be nice to mention about that in _Controller not working with shouldFocusError? (V6)_ section of [faqs](https://react-hook-form.com/faqs).

Thanks for your time! [🙏](https://emojipedia.org/folded-hands/)

**Reply by Phe0** [reply]
Hello, I'm trying  to achieve a behavior in which the page would scroll to the input without focusing it. The validation mode on my application is set to 'onBlur', so focusing it would make the errors vanish. Would there be any way to achieve this behavior?

**Reply by johnandresmedina** [reply]
> Well, I have found the cause of my problem why some fields are not focused by the default lib mechanism. I had to add `tabIndex` to root divs of my components. I think that it would be nice to mention about that in _Controller not working with shouldFocusError? (V6)_ section of [faqs](https://react-hook-form.com/faqs).
> 
> Thanks for your time! [🙏](https://emojipedia.org/folded-hands/)

I'm facing the exact same issue, wondering how did you solve it?
I'm using the latest version of the library: "react-hook-form": "7.62.0"

**Reply by terrymun** [reply]
We ran into the same issue here: it seems that instead of passing the entire element in, the `ref` is only wired up to expose specific APIs from the element when using a controller:

https://github.com/react-hook-form/react-hook-form/blob/37f51ac10b49d45f73871d388db7312a50b4cae4/src/useController.ts#L161-L169

For our use-case, we would like to have access to the underlying element in order to get its associated labels, in order to generate a error summary containing the labels of the invalid field(s).

We ended up having to perform a manual patch to override this behavior:

```ts
if (field && elm) {
  field._f.ref = elm;
}
```
